src/v2/mod.rs

```diff
@@ -65,74 +59,52 @@ pub struct Client {
     token: Option<String>,
 }
 
-/// Convenience alias for an arbitrary future type
-pub type FutureResult<T> = Box<dyn Future<Item = T, Error = Error> + Send>;
-
-/// Convenience alias for a future boolean result.
-pub type FutureBool = Box<dyn Future<Item = bool, Error = Error> + Send>;
-
-/// Convenience alias for a future manifest blob.
-pub type FutureManifest = Box<dyn Future<Item = Manifest, Error = Error> + Send>;
-
-/// Convenience alias for a future manifest blob and ref.
-pub type FutureManifestAndRef =
-    Box<dyn Future<Item = (Manifest, Option<String>), Error = Error> + Send>;
-
 impl Client {
     pub fn configure() -> Config {
         Config::default()
     }
 
     /// Ensure remote registry supports v2 API.
-    pub fn ensure_v2_registry(self) -> impl Future<Item = Self, Error = Error> + Send {
-        self.is_v2_supported()
-            .map(move |ok| (ok, self))
-            .and_then(|(ok, client)| {
-                if !ok {
-                    bail!("remote server does not support docker-registry v2 API")
-                } else {
-                    Ok(client)
-                }
-            })
+    pub async fn ensure_v2_registry(self) -> Result<Self> {
```

Comment:
I think this can be changed to a simpler `&self` and `-> Result<()>`.

Reply:
Unless you feel strongly about this I'd rather not change the signature like that, as it'll brake consumers who might use this in a chained fashion.